chain/transaction_test.go

```diff
@@ -252,3 +334,146 @@ func TestSignRawActionBytesTx(t *testing.T) {
 	require.NoError(err)
 	require.Equal(signedTx.Bytes(), rawSignedTxBytes)
 }
+
+func TestPreExecute(t *testing.T) {
```

Comment:
(could be split to separate PR) - we should write unit tests for the pre executor type itself as well, since that is how `PreExecute` will be called from outside this package